src/_1_getting_started/_2_1_hello_triangle.rs

```diff
@@ -66,7 +66,7 @@ pub fn main_1_2_1() {
 
         // check for shader compile errors
         let mut success = gl::FALSE as GLint;
-        let mut infoLog = Vec::with_capacity(512);
+        let mut infoLog = vec![0; 512];
         infoLog.set_len(512 - 1); // subtract 1 to skip the trailing null character
```

Comment:
I guess this line is made obsolete by the change?

Reply:
Removing this line would include the null terminator which works just as fine. Your comment got me thinking though, I modified the PR to set the length the proper way by sending a mutable pointer to `gl::GetShaderInfoLog` to obtain the length of the string data copied to the `infoLog` buffer.